daemon/api_test.go

```diff
@@ -1336,22 +1336,34 @@ func (s *apiSuite) TestConnectPlugSuccess(c *check.C) {
 		"status_code": 200.0,
 		"type":        "sync",
 	})
-	for slot, plugs := range d.interfaces.ConnectedSlots("consumer") {
-		c.Check(slot.Snap, check.Equals, "consumer")
-		c.Check(slot.Name, check.Equals, "slot")
-		for _, plug := range plugs {
-			c.Check(plug.Snap, check.Equals, "producer")
-			c.Check(plug.Name, check.Equals, "plug")
-		}
-	}
-	for plug, slots := range d.interfaces.ConnectedPlugs("producer") {
-		c.Check(plug.Snap, check.Equals, "producer")
-		c.Check(plug.Name, check.Equals, "plug")
-		for _, slot := range slots {
-			c.Check(slot.Snap, check.Equals, "consumer")
-			c.Check(slot.Name, check.Equals, "slot")
-		}
-	}
+	c.Assert(d.interfaces.Interfaces(), check.DeepEquals, &interfaces.Interfaces{
+		Plugs: []*interfaces.Plug{
+			&interfaces.Plug{
```

Comment:
I actually already have a patch that makes it into plain non-pointer objects. I wasn't sure if I want to include it so that I can move away from the refactorings and get back to the core business. I can post it quickly if you want :)